code/modules/mob/living/carbon/xenomorph/abilities/predalien/predalien_powers.dm

```diff
@@ -42,116 +39,183 @@
 	apply_cooldown()
 	return ..()
 
-/datum/action/xeno_action/onclick/smash/use_ability(atom/target)
+/datum/action/xeno_action/activable/feralfrenzy/use_ability(atom/target)
 	var/mob/living/carbon/xenomorph/xeno = owner
+	if(!action_cooldown_check() || xeno.action_busy)
+		return
+	if(!xeno.check_state())
+		return
+
+	var/aoe_damage = base_damage_aoe
+	var/aoe_scale = damage_scale_aoe
```

Comment:
These two aren't needed if you aren't going to modify them and it makes the code no more readable. Use base_damage_aoe and damage_scale_aoe.